receiver/awsfirehosereceiver/config.go

```diff
@@ -0,0 +1,27 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package awsfirehosereceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/awsfirehosereceiver"
+
+import (
+	"go.opentelemetry.io/collector/config"
+	"go.opentelemetry.io/collector/config/confighttp"
+)
+
+type Config struct {
+	config.ReceiverSettings       `mapstructure:",squash"`
+	confighttp.HTTPServerSettings `mapstructure:",squash"`
+	Encoding                      string `mapstructure:"encoding"`
+	AccessKey                     string `mapstructure:"access_key"`
+}
```

Comment:
If an unknown `Encoding` value is specified, then the receiver creation will return an error and fail. This way the correct unmarshaler set can be checked, so if a metrics `Encoding` value is specified for a trace pipeline, it will fail.

Reply:
Checking in `Config.Validate()` will enable failing early as well as planned-but-not-implemented features like a config linter or dry run.  Better to die when parsing the config than waiting until a pipeline is constructed.

Reply:
I'll probably have it die in both. Definitely need to make sure the encoding is going to work with the pipeline.